### PR TITLE
Fix correctness mistakes

### PR DESCRIPTION
**Pull request by vinnik-dmitry07**

### CODE REVIEW

File: doc/readme/ABOUT.md [hunk 1]
```diff
@@ -2,16 +2,16 @@ Shogun is an open-source machine learning library that offers a wide range of ef
...
 #### Shogun is accessible
 
-* Supports many languages (Python, Octave, R, Java/Scala, Lua, C#, Ruby, etc) and platforms (Linux/Unix, MacOS and Windows) and integrates with their scientific computing environments.
+* Supports many languages (Python, Octave, R, Java/Scala, Lua, C#, Ruby, etc) and platforms (Linux/Unix, macOS, and Windows) and integrates with their scientific computing environments.
 
 #### Shogun is state-of-the-art
 
 * Efficient implementation (from standard to cutting edge algorithms), modern software architecture in C++.
-* Easy combination of multiple data representations, algorithm classes and general purpose tools for rapid prototyping of data pipelines.
+* Easy combination of multiple data representations, algorithm classes, and general-purpose tools for rapid prototyping of data pipelines.
```
Comment (gf712) [inline doc/readme/ABOUT.md:10]: I guess the Oxford comma is a matter of opinion :D

Comment (vinnik-dmitry07) [reply doc/readme/ABOUT.md:10]: I agree with you but there are some pros to do this: the first is GitHub's use of Oxford comma (even in the frequently observed message _"Learn Git and GitHub without any code!
Using the Hello World guide, you’ll start a branch, write comments, and open a pull request."_, the second is having some number of sentences with the comma in the docs e.g. https://github.com/shogun-toolbox/shogun/pull/4906/files#diff-f5f6b994e6c7890608d6a4dfacbef362L105

Comment (gf712) [reply doc/readme/ABOUT.md:10]: true! I don't have an opinion on it tbh, just an observation :)

**Comment by vigsterkr** [comment]
if there are no objections i'll merge this soon....

File: README.md [hunk 2]
```diff
@@ -63,7 +63,7 @@ Note that some folders are submodules that can be checked out with
 `git submodule update --init`.
 
 - *src* - source code, separated into C++ source and interfaces
-- *doc* - readmes (doc/reamde, submodule), ipython notebooks, cookbook (api examples), licenses
+- *doc* - readmes (doc/reamdme, submodule), Jupyter notebooks, cookbook (API examples), licenses
```
Comment (gf712) [inline README.md]:
```suggestion
- *doc* - readmes (doc/readme, submodule), Jupyter notebooks, cookbook (API examples), licenses
```

File: doc/readme/DEVELOPING.md [hunk 3]
```diff
@@ -95,14 +95,14 @@ Shogun has a custom script called `check_format.sh` which can be used to verify
  * The PR is small in terms of lines changes.
  * The PR is clean and addresses **one** issue.
  * The number of commits is minimal (i.e. one), the message is neat and clear.
- * If C++ code: it is covered by [tests](#testing), it doesn't leak memory, its [API](#api) is documented, [code style](https://github.com/shogun-toolbox/shogun/wiki/Code-style).
- * If API example: it has a clear scope, it is minimal, it looks polished, it has a passing [test](#testing)
+ * If a C++ code: it is covered by [tests](#testing), it doesn't leak memory, its [API](#api) is documented, [code style](https://github.com/shogun-toolbox/shogun/wiki/Code-style).
```
Comment (gf712) [inline doc/readme/DEVELOPING.md]:
```suggestion
 * For C++ code: it is covered by [tests](#testing), it doesn't leak memory, its [API](#api) is documented, [code style](https://github.com/shogun-toolbox/shogun/wiki/Code-style).
```

File: doc/readme/DEVELOPING.md [hunk 4]
```diff
@@ -95,14 +95,14 @@ Shogun has a custom script called `check_format.sh` which can be used to verify
...
+ * If an API example: it has a clear scope, it is minimal, it looks polished, it has a passing [test](#testing)
  * If docs: clear, correct English language, spell-checked
- * If notebook: cell output is removed, template is respected, plots have axis labels.
+ * If a notebook: the cell output is removed, the template is respected, the plots have axis labels.
  * Formatting notebooks/docs: Please every sentence in a single line.
 
 # Testing <a name="testing"></a>
-There are three types of tests that can be executed locally, C++ unit tests, running the API examples, and integration testing the results of the API examples.
+Three types of tests can be executed locally, C++ unit tests, running the API examples, and integration testing the results of the API examples.
```
Comment (gf712) [inline doc/readme/DEVELOPING.md]:
```suggestion
Three types of tests can be executed locally: C++ unit tests, running the API examples, and integration testing the results of the API examples.
```

File: doc/readme/DEVELOPING.md [hunk 5]
```diff
@@ -148,16 +148,16 @@ Then
 
 The option `--leak-check=full` for valgrind might be useful.
 In addition to manually running valgrind on your tests, you can use `ctest` to check multiple tests.
-This requires to be enable in dashboard reports in via `-DBUILD_DASHBOARD_REPORTS=ON`.
+This requires to be enabled in dashboard reports in via `-DBUILD_DASHBOARD_REPORTS=ON`.
```
Comment (gf712) [inline doc/readme/DEVELOPING.md]:
```suggestion
This requires to be enabled in dashboard reports with `-DBUILD_DASHBOARD_REPORTS=ON`.
```

File: doc/readme/DEVELOPING.md [hunk 6]
```diff
@@ -148,16 +148,16 @@ Then
...
 
     ctest -D ExperimentalMemCheck -R unit-GaussianProcessRegression
 
 #### Adding tests
 We aim to write clear, minimal, yet exhaustive tests of basic building blocks in Shogun.
-Whenever you send us C++ code, we will ask you for a unit test for it.
+Whenever you send us a C++ code, we will ask you for a unit test for it.
 We do test numerical results as compared to reference implementations (e.g. in Python), as well as corner cases, consistency etc.
-Read on [test driven development](https://en.wikipedia.org/wiki/Test-driven_development), and search the web for tips on unit tests, e.g. [googletest's tips](https://github.com/google/googletest/blob/master/googletest/docs/Primer.md).
+Read on [test-driven development](https://en.wikipedia.org/wiki/Test-driven_development), and search the web for tips on unit tests, e.g. [googletest's tips](https://github.com/google/googletest/blob/master/googletest/docs/Primer.md).
```
Comment (gf712) [inline doc/readme/DEVELOPING.md]: or TDD, as most people seem to call it now

File: doc/readme/DEVELOPING.md [hunk 7]
```diff
@@ -234,34 +234,34 @@ If everything worked, then the [travis](#buildfarm) build in the second PR will
...
     ctest -L benchmark
 
 ## Adding benchmarks
 We aim to provide an easy way to benchmark modules in Shogun. Hence, whenever you send us new C++ implementation, please
 consider writing benchmarks for it.
 
-In order to add benchmark to a class, simply create a file next to the implementation, which name is suffixed with `_benchmark.cc`.
-For example for the `RandomFourierDotFeatures` class the benchmarks are available in the `RandomFourierDotFeatures_benchmark.cc` file.
-On top of this you need to explicitly specify to cmake that you have added a new benchmarking file.
+To add benchmark to a class, simply create a file next to the implementation, which name is suffixed with `_benchmark.cc`.
```
Comment (gf712) [inline doc/readme/DEVELOPING.md]:
```suggestion
To add a benchmark to a class, simply create a file next to the implementation, whose name is suffixed with `_benchmark.cc`.
```

File: doc/readme/DEVELOPING.md [hunk 8]
```diff
@@ -234,34 +234,34 @@ If everything worked, then the [travis](#buildfarm) build in the second PR will
...
+For example, for the `RandomFourierDotFeatures` class the benchmarks are available in the `RandomFourierDotFeatures_benchmark.cc` file.
+On top of this, you need to explicitly specify to cmake that you have added a new benchmarking file.
```
Comment (gf712) [inline doc/readme/DEVELOPING.md]:
```suggestion
On top of this, you need to explicitly specify in CMakeLists.txt that you have added a new benchmarking file.
```

File: doc/readme/DEVELOPING.md [hunk 9]
```diff
@@ -234,34 +234,34 @@ If everything worked, then the [travis](#buildfarm) build in the second PR will
...
 `ADD_SHOGUN_BENCHMARK` cmake helper function should be used in `src/shogun/CMakeLists.txt` within the `benchmarking` to add a
```
Comment (gf712) [inline doc/readme/DEVELOPING.md]:
```suggestion
The `ADD_SHOGUN_BENCHMARK` cmake helper function should be used in `src/shogun/CMakeLists.txt` to add a
```

File: doc/readme/DEVELOPING.md [hunk 10]
```diff
@@ -234,34 +234,34 @@ If everything worked, then the [travis](#buildfarm) build in the second PR will
...
-benchmark implementations to build.
+benchmark implementation to build.
```
Comment (gf712) [inline doc/readme/DEVELOPING.md]:
```suggestion
benchmark implementation.
```

File: doc/readme/DEVELOPING.md [hunk 11]
```diff
@@ -234,34 +234,34 @@ If everything worked, then the [travis](#buildfarm) build in the second PR will
...
 a new benchmarking file, add new benchmarks with the
```
Comment (gf712) [inline doc/readme/DEVELOPING.md]:
```suggestion
```

File: doc/readme/DEVELOPING.md [hunk 12]
```diff
@@ -234,34 +234,34 @@ If everything worked, then the [travis](#buildfarm) build in the second PR will
...
 `ADD_SHOGUN_BENCHMARK` cmake function the following way. In the `src/shogun/CMakeLists.txt`
```
Comment (gf712) [inline doc/readme/DEVELOPING.md]:
```suggestion
For example, to add a benchmark with the path `src/classifier/YOLOClassifier_benchmark.cc`, use the `ADD_SHOGUN_BENCHMARK` cmake function in the following way in `src/shogun/CMakeLists.txt`
```

File: doc/readme/DEVELOPING.md [hunk 13]
```diff
@@ -234,34 +234,34 @@ If everything worked, then the [travis](#buildfarm) build in the second PR will
...
-file within the `benchmarking` section using th
+file within the `benchmarking` section using the
```
Comment (gf712) [inline doc/readme/DEVELOPING.md]:
```suggestion
file within the `benchmarking` section:
```

**Comment by gf712** [comment]
> if there are no objections i'll merge this soon....

added a few comments, but didn't go through everything

File: doc/readme/DEVELOPING.md (hunk 6 above)
Comment (vinnik-dmitry07) [reply doc/readme/DEVELOPING.md]: Incorporate it.

**Comment by vinnik-dmitry07** [comment]
That is it.

**Comment by vigsterkr** [comment]
@gf712 squash'n'merge?
